source/linux/background-managers/gnome.js

```diff
@@ -4,21 +4,42 @@ export async function isAvailable() {
 	return commandExists('gsettings');
 }
 
+export async function isDarkTheme() {
+	const {themeStyle} = await execFile('gsettings', [
+		'get',
+		'org.gnome.desktop.interface',
+		'color-scheme',
+	]);
+
+	return Boolean(themeStyle === 'prefer-dark');
+}
+
 export async function get() {
+	const prop = (isDarkTheme()) ? 'picture-uri-dark' : 'picture-uri';
+
 	const {stdout} = await execFile('gsettings', [
 		'get',
 		'org.gnome.desktop.background',
-		'picture-uri',
+		prop,
 	]);
 
 	return stdout.trim().slice(8, -1);
 }
 
 export async function set(imagePath) {
-	await execFile('gsettings', [
-		'set',
-		'org.gnome.desktop.background',
-		'picture-uri',
-		`file://${imagePath}`,
-	]);
+	if (isDarkTheme()) {
+		await execFile('gsettings', [
+			'set',
+			'org.gnome.desktop.background',
+			'picture-uri-dark',
```

Comment:
The only difference is this, so it can just be a ternary here instead of an `if`.

Reply:
I changed it on my last commit